tools/wasm-sourcemap.py

```diff
@@ -231,15 +233,183 @@ def extract_comp_dir_map(text):
   return map_stmt_list_to_comp_dir
 
 
-def read_dwarf_entries(wasm, options):
+def demangle_names(names):
+  # Only demangle names that look mangled
+  mangled_names = sorted({n for n in names if n.startswith('_Z')})
+  if not mangled_names:
+    return {}
+  if not os.path.exists(LLVM_CXXFILT):
+    logger.warning('llvm-cxxfilt does not exist')
+    return {}
+
+  # Gather all mangled names and call llvm-cxxfilt only once for all of them
+  try:
+    input_str = '\n'.join(mangled_names)
+    process = Popen([LLVM_CXXFILT], stdin=PIPE, stdout=PIPE, stderr=PIPE, text=True)
+    stdout, stderr = process.communicate(input=input_str)
+    if process.returncode != 0:
+      logger.warning('llvm-cxxfilt failed: %s' % stderr)
+      return {}
+
+    demangled_list = stdout.splitlines()
+    if len(demangled_list) != len(mangled_names):
+      logger.warning('llvm-cxxfilt output length mismatch')
+      return {}
+
+    return dict(zip(mangled_names, demangled_list))
+  except OSError:
+    logger.warning('Failed to run llvm-cxxfilt')
+    return {}
```

Comment:
You can skip the whole try/except here if you use `shared.check_call`

Reply:
Done: https://github.com/emscripten-core/emscripten/pull/25870/commits/c8442567c45ce4613e5b4c6ff749bff2e41b749b